controllers/component/predicates.go

```diff
@@ -5,8 +5,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
+// ComponentCreatedPredicate returns a predicate which filters out
+// only components that are the result of a create event
+func ComponentCreatedPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(genericEvent event.GenericEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
```

Comment:
We may need to also `return true` here since we will check the `DeletionTimestamp`

Reply:
Sorry, ignore this comment, this function is for create event only.